### PR TITLE
Expose required S.Resources members on corelib

### PR DESCRIPTION
**Pull request by alexperovich**
Fixes dotnet/corefx#12012

### CODE REVIEW

**Comment by alexperovich** [comment]
@jkotas

**Comment by jkotas** [comment]
LGTM

File: src/mscorlib/src/System/Resources/ResourceSet.cs [hunk 1]
```diff
@@ -182,15 +182,17 @@ public virtual Type GetDefaultReader()
...
         }
     
-#if !FEATURE_CORECLR
         // Returns the preferred IResourceWriter class for this kind of ResourceSet.
         // Subclasses of ResourceSet using their own Readers &; should override
         // GetDefaultReader and GetDefaultWriter.
         public virtual Type GetDefaultWriter()
         {
+#if FEATURE_CORECLR
+            return Type.GetType("System.Resources.ResourceWriter, System.Resources.Writer, Version=4.0.1.0, Culture=neutral, PublicKeyToken=" + AssemblyRef.MicrosoftPublicKeyToken, throwOnError: true);
```
Comment (jkotas) [inline src/mscorlib/src/System/Resources/ResourceSet.cs]: Do we also need to expose `IResourceWriter` interface from here for the ResourceWriter to implement? Note that there is code in Reflection.Emit that calls IResourceWriter.

I would be useful to physically exclude ResourceWriter from the CoreCLR build to make sure that it is not leaking anywhere.

Comment (alexperovich) [inline src/mscorlib/src/System/Resources/ResourceSet.cs]: The ResourceWriter.cs file isn't even in the repository, IResourceWriter is though. The interface and type are both in corefx. If other public things in corelib expose IResourceWriter then we need to expose it as well.

**Pull request merged**